Issue45: fix student page

Hellomyguest opened the pull request.

Commit:
feat(components): add table for homeworks

META DATA: issue45

Commit:
refactor(components): student page and student settings page refactored

META DATA: issue45

dpolevodin commented on src/features/student-links/StudentLinks.tsx:
```diff
@@ -1,15 +1,16 @@
 type Props = {
-  tgNickName: string;
-  githubNickName: string;
+  tgNickName: string | undefined;
```
если свойство может быть undefined, то можно попробовать сделать запись короче
```
tgNickName?: string;
```

Hellomyguest replied:
Поправил, пришлось только defaultProps добавить.

dpolevodin commented on src/pages/sign/signUp/lib/types.ts:
```diff
@@ -0,0 +1,28 @@
+        id?: number;
+        title: string;
+        deadline?: string;
+        status?: "default" | "approved" | "pending" | "rejected";
```
думаю стоит попробовать вынести статусы в отдельную  enum и сюда уже передавать enum
Пример 
```
export enum HomeWorsStatus {
  DEFAULT = 'default',
  APPROVED = 'approved',
  ...
}

...
status?: HomeWorsStatus;
...
```

Hellomyguest replied:
Но ведь в таком случае в я не смогу использовать в объекте домашних заданий строковые значения?
Пример : 
`homework1: {
          id: 1,
          title: "ДЗ 1",
          deadline: "01.01.2023",
          status: "approved",
        },`

будет ошибка

dpolevodin replied:
в данном случае можно вставлять также значение из enum, так как он ждет не string, а вариант из enum 
`homework1: { id: 1, title: "ДЗ 1", deadline: "01.01.2023", status: HomeWorsStatus.APPROVED },`
Попробуй, пожалуйста, так, ошибки быть не должно
и в будущем, ты все статусы по проекту сможешь в одной enum хранить

Hellomyguest replied:
готово

Commit:
feat(components): add spin to pagelayout

getSession moved from homepage to app
add spin to pagelayout
fix navigation sider flashing

META DATA: issue45

Commit:
Merge branch 'develop' into issue45/fix-student-page

Commit:
fix(components): fix key error and add enum for homeworks statuses

META DATA: issue45
